### PR TITLE
Tron 39 tron architecture fixup

### PR DESCRIPTION
**Pull request by twtaylorbitgo**
Architecture fixup for Tron.

This is based on comments left in https://github.com/BitGo/bitgo-account-lib/pull/1 and also general architecture redesign. This removes addFrom and addDestination.

### CODE REVIEW

File: src/coin/baseCoin/errors.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export class ExtendableError extends Error {
 export class ParseTransactionError extends ExtendableError {
   constructor(message: string) {
     super(message);
-    this.name = 'TransactionParseError';
+    this.name = 'ParseTransactionError';
```
Comment (OttoAllmendinger) [inline src/coin/baseCoin/errors.ts]: `ExtendableError` can just use `this.constructor.name`

Comment (twtaylor) [reply src/coin/baseCoin/errors.ts]: ah good point

File: src/coin/trx/enum.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export enum ContractType {
-  Transfer,
+  Transfer = 1,
```
Comment (argjv) [inline src/coin/trx/enum.ts]: What does the `1` mean?

File: src/coin/trx/enum.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export enum ContractType {
```
Comment (argjv) [inline src/coin/trx/enum.ts]: jsdocs

File: src/coin/trx/utils.ts [hunk 4]
```diff
@@ -104,26 +109,25 @@ export function decodeTransaction(base64EncodedHex: string): RawTransaction {
 
 /**
  * Decodes a transaction's raw field from a base64 encoded string. This is a protobuf representation.
- * @param base64EncodedHex
+ * @param hexString
  * @example
  * @see {@link }
```
Comment (argjv) [inline src/coin/trx/utils.ts]: Update jsdoc, missing link and example

File: src/coin/trx/utils.ts [hunk 5]
```diff
@@ -132,15 +136,7 @@ export function decodeRawTransaction(base64EncodedHex: string): { expiration: nu
...
-  if (!hex.match(/^(0x)?[0-9a-fA-F]*$/)) {
-    return false;
-  }
-
-  if ((hex.length % 2) !== 0) {
-    return false;
-  }
-
-  return true;
+  return /^(0x)?([0-9a-f]{2})+$/i.test(hex);
```
Comment (argjv) [inline src/coin/trx/utils.ts]: Need a test for this. I've seen too many regex that doesn't work.

File: src/transactionBuilder.ts [hunk 6]
```diff
@@ -10,15 +10,13 @@ export interface TransactionBuilderParams {
   coinName?: string;
 }
 
-export default class TransactionBuilder {
-  private transaction: BaseTransaction;
+export class TransactionBuilder {
+  public transaction: BaseTransaction;
```
[the remark this answers is not in the document]
Comment (twtaylorbitgo) [reply src/transactionBuilder.ts]: eh we can keep this private, I wanted to look at as it comes back in test. maybe we can just changed this in test coin? or surface it through a getter?

File: src/coin/trx/enum.ts (hunk 3 above)
Comment (twtaylorbitgo) [reply src/coin/trx/enum.ts]: ack

File: src/coin/trx/enum.ts (hunk 2 above)
Comment (twtaylorbitgo) [reply src/coin/trx/enum.ts]: i've removed it. I jsut didn't want to look for 0 but figure that's fine

File: src/coin/trx/utils.ts (hunk 5 above)
Comment (twtaylorbitgo) [reply src/coin/trx/utils.ts]: fair enough, added one

File: src/coin/trx/utils.ts (hunk 4 above)
Comment (twtaylorbitgo) [reply src/coin/trx/utils.ts]: done

**Pull request merged**